app/assets/javascripts/change_current_provider.coffee

```diff
@@ -4,6 +4,13 @@ $(document).ready ->
     overlay: 0.6
     closeButton: '.modal-close'
 
+  $('a.modal-close-link').on 'click', (element) ->
```

Comment:
This is necessary because using modal-close causes the button to float right because of the CSS on lines 33-36 of _modal.scss  I think I would favor a solution where you overwrite the CSS and use modal-close rather than this JS.